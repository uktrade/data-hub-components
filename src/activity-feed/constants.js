export const SOURCE_TYPES = {
  external: 'externalDataSource',
}

export const ACTIVITY_TYPE = {
  CompaniesHouseAccount: ['dit:Accounts'],
  CompaniesHouseCompany: ['dit:Company'],
  HmrcExporter: ['dit:Export'],
  Interaction: ['dit:Interaction', 'dit:ServiceDelivery'],
  InvestmentProject: ['dit:InvestmentProject'],
  Omis: ['dit:OMISOrder'],
}

export const ACTIVITY_TYPE_FILTERS = {
  allActivity: {
    label: 'All Data Hub & external activity',
    value: 'allActivity',
  },
  myActivity: {
    label: 'My activity',
    value: 'myActivity',
  },
  externalActivity: {
    label: 'All external activity',
    value: 'externalActivity',
  },
  dataHubActivity: {
    label: 'All Data Hub activity',
    value: 'dataHubActivity',
  },
}

export default {
  ACTIVITY_TYPE,
  ACTIVITY_TYPE_FILTERS,
  SOURCE_TYPES,
}
