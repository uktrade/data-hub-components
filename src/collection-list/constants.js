const MAX_ITEMS_TO_DOWNLOAD = 5000

export default MAX_ITEMS_TO_DOWNLOAD
