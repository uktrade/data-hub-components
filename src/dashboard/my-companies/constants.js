export const LIST_CHANGE = 'LIST_CHANGE'
export const FILTER_CHANGE = 'FILTER_CHANGE'
export const ORDER_CHANGE = 'ORDER_CHANGE'
