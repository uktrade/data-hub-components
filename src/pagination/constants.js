export const PAGINATION_PIECE_PREVIOUS = 'previous'
export const PAGINATION_PIECE_ELLIPSIS = 'ellipsis'
export const PAGINATION_PIECE_PAGE_NUMBER = 'page-number'
export const PAGINATION_PIECE_NEXT = 'next'

export const DEFAULT_MAX_PAGE_NUMBER_LINKS = 5
