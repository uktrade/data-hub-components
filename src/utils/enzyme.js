export const withTargetValue = value => ({
  target: {
    value,
  },
})

/**
 * @function getOptionValueByText
 * @description Finds an `<option>` element with its inner text matching the
 * {text} parameter and returns the value of its `value` attribute.
 * @param {*} wrapper - Enzyme React wrapper
 * @param {String} text - The option text
 */
export const getOptionValueByText = (wrapper, text) => (
  wrapper.find('option')
    .find({ children: text })
    .props()
    .value
)

/**
 * @function changeAndUpdate
 * @description Simulates the `change` event on the provided {wrapper} with
 * {vaule} as `target.value` and updates (re-renders) the wrapper.
 * @param {*} wrapper - Enzyme React wrapper
 * @param {*} value - The value to be used as `event.target`
 */
export const changeAndUpdate = (wrapper, value) => {
  wrapper.simulate('change', withTargetValue(value))
  wrapper.update()
}

/**
 * @function changeSelectAndUpdate
 * @description Finds a `<select>` element in the wrapper, selects the option
 * matching the {text} and updates (re-renders) the wrapper.
 * @param {*} wrapper - Enzyme React wrapper
 * @param {String} text - The option text
 */
export const changeSelectAndUpdate = (wrapper, text) => {
  changeAndUpdate(wrapper.find('select'), getOptionValueByText(wrapper, text))
  wrapper.update()
}
