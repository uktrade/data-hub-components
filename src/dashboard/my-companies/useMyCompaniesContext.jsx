import { orderBy, pick } from 'lodash'
import { useReducer } from 'react'
import createUseContext from 'constate'
import { FILTER_CHANGE, LIST_CHANGE, ORDER_CHANGE } from './constants'

export const filterCompanyName = (companies, filterText) =>
  filterText.length
    ? companies.filter((c) =>
        c.company.name.toLowerCase().includes(filterText.toLowerCase())
      )
    : companies

const defaultState = {
  lists: [],
  selectedIdx: 0,
  sortBy: 'recent',
  filter: '',
}

export const reducer = (state, { type, ...action }) => {
  switch (type) {
    case LIST_CHANGE:
      return { ...state, selectedIdx: action.idx }
    case FILTER_CHANGE:
      return { ...state, filter: action.filter }
    case ORDER_CHANGE:
      return { ...state, sortBy: action.sortBy }
    default:
      return state
  }
}

// We are unpacking children here just to remove them from initialState.
const useMyCompaniesContext = createUseContext(
  ({ children, editListsLinkProps, ...rest }) => {
    const initialState = pick(rest, [
      'lists',
      'selectedIdx',
      'sortBy',
      'filter',
    ])
    const [state, dispatch] = useReducer(reducer, {
      ...defaultState,
      ...initialState,
      // Reducer assumes the list of company lists is sorted.
      lists: orderBy(initialState.lists, 'name'),
    })

    return {
      state,
      dispatch,
      editListsLinkProps,
    }
  }
)

export default useMyCompaniesContext
