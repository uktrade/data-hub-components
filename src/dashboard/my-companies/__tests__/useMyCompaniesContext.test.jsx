import _ from 'lodash'
import React from 'react'
import { mount } from 'enzyme'
import useMyCompaniesContext, {
  getSortedCompanies,
  reducer,
  filterCompanyName,
} from '../useMyCompaniesContext'
import companies from '../../__fixtures__/companies'
import { LIST_CHANGE, FILTER_CHANGE, ORDER_CHANGE } from '../constants'

const initialState = {
  lists: [
    { name: 'Foo' },
    { name: 'Bar' },
    { name: 'Baz' },
  ],
  selectedIdx: 0,
  sortBy: 'alphabetical',
  filter: 'foobarbaz',
}

describe('Store', () => {
  describe('getSortedCompanies()', () => {
    test('sort by recent', () => {
      const expected = [
        {
          company: {
            name: 'A1 BMW LTD!!!!!',
            id: 'a1138c1c-d449-4846-aa58-18fae7e1cb92',
            isArchived: false,
          },
          latestInteraction: {
            id: '79d92719-7402-45b6-b3d7-eff559d6b282',
            date: '2019-08-14',
            subject:
              'Here is a long interaction title some more text some more text some more text almost finished some more text nearly there more text finished',
          },
        },
        {
          company: {
            name: 'Zebra clothing',
            id: 'b89b1db3-7140-44ca-ad7a-9824c3c2gh74',
            isArchived: false,
          },
          latestInteraction: {
            id: '86f92719-7402-45b6-b3d7-eff559d6b678',
            date: '2019-02-21',
            subject:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
          },
        },
        {
          company: {
            name: 'Portable Potatoes',
            id: 'a30b1db3-7140-44ca-ad7a-9824c3c2ed56',
            isArchived: false,
          },
          latestInteraction: {
            date: null,
            subject: 'No interactions have been recorded',
          },
        },
      ]
      const recent = getSortedCompanies(companies, 'recent')
      expect(recent).toEqual(expected)
    })
    test('sort by least recent', () => {
      const expected = [
        {
          company: {
            name: 'Portable Potatoes',
            id: 'a30b1db3-7140-44ca-ad7a-9824c3c2ed56',
            isArchived: false,
          },
          latestInteraction: {
            date: null,
            subject: 'No interactions have been recorded',
          },
        },
        {
          company: {
            name: 'Zebra clothing',
            id: 'b89b1db3-7140-44ca-ad7a-9824c3c2gh74',
            isArchived: false,
          },
          latestInteraction: {
            id: '86f92719-7402-45b6-b3d7-eff559d6b678',
            date: '2019-02-21',
            subject:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
          },
        },
        {
          company: {
            name: 'A1 BMW LTD!!!!!',
            id: 'a1138c1c-d449-4846-aa58-18fae7e1cb92',
            isArchived: false,
          },
          latestInteraction: {
            id: '79d92719-7402-45b6-b3d7-eff559d6b282',
            date: '2019-08-14',
            subject:
              'Here is a long interaction title some more text some more text some more text almost finished some more text nearly there more text finished',
          },
        },
      ]
      const leastRecent = getSortedCompanies(companies, 'least-recent')
      expect(leastRecent).toEqual(expected)
    })
    test('sort by alphabetical', () => {
      const expected = [
        {
          company: {
            name: 'A1 BMW LTD!!!!!',
            id: 'a1138c1c-d449-4846-aa58-18fae7e1cb92',
            isArchived: false,
          },
          latestInteraction: {
            id: '79d92719-7402-45b6-b3d7-eff559d6b282',
            date: '2019-08-14',
            subject:
              'Here is a long interaction title some more text some more text some more text almost finished some more text nearly there more text finished',
          },
        },
        {
          company: {
            name: 'Portable Potatoes',
            id: 'a30b1db3-7140-44ca-ad7a-9824c3c2ed56',
            isArchived: false,
          },
          latestInteraction: {
            date: null,
            subject: 'No interactions have been recorded',
          },
        },
        {
          company: {
            name: 'Zebra clothing',
            id: 'b89b1db3-7140-44ca-ad7a-9824c3c2gh74',
            isArchived: false,
          },
          latestInteraction: {
            id: '86f92719-7402-45b6-b3d7-eff559d6b678',
            date: '2019-02-21',
            subject:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
          },
        },
      ]
      const alphabetical = getSortedCompanies(companies, 'alphabetical')
      expect(alphabetical).toEqual(expected)
    })
  })
  describe('filterCompanyName()', () => {
    test('Filters list of companies by name', () => {
      const expected = [
        {
          company: {
            name: 'Portable Potatoes',
            id: 'a30b1db3-7140-44ca-ad7a-9824c3c2ed56',
            isArchived: false,
          },
          latestInteraction: {
            date: null,
            subject: 'No interactions have been recorded',
          },
        },
      ]
      const recent = filterCompanyName(companies, 'portable')
      expect(recent).toEqual(expected)
    })
    test('returns all companies if filter value is empty', () => {
      const recent = filterCompanyName(companies, '')
      expect(recent).toEqual(companies)
    })
  })
  describe('reducer()', () => {
    test('List change', () => {
      expect(reducer(initialState, {
        type: LIST_CHANGE,
        idx: 1,
      })).toEqual({ ...initialState, selectedIdx: 1 })

      expect(reducer(initialState, {
        type: LIST_CHANGE,
        idx: 2,
      })).toEqual({ ...initialState, selectedIdx: 2 })

      expect(reducer(initialState, {
        type: LIST_CHANGE,
        idx: 0,
      })).toEqual({ ...initialState, selectedIdx: 0 })
    })

    test('Order by change', () => {
      expect(reducer(initialState, {
        type: ORDER_CHANGE,
        sortBy: 'recent',
      })).toEqual({ ...initialState, sortBy: 'recent' })

      expect(reducer(initialState, {
        type: ORDER_CHANGE,
        sortBy: 'least-recent',
      })).toEqual({ ...initialState, sortBy: 'least-recent' })

      expect(reducer(initialState, {
        type: ORDER_CHANGE,
        sortBy: 'alphabetical',
      })).toEqual({ ...initialState, sortBy: 'alphabetical' })
    })

    test('Filter change', () => {
      expect(reducer(initialState, {
        type: FILTER_CHANGE,
        filter: 'foo',
      })).toEqual({ ...initialState, filter: 'foo' })

      expect(reducer(initialState, {
        type: FILTER_CHANGE,
        filter: 'bar',
      })).toEqual({ ...initialState, filter: 'bar' })

      expect(reducer(initialState, {
        type: FILTER_CHANGE,
        filter: '',
      })).toEqual({ ...initialState, filter: '' })
    })
  })

  describe('useMyCompaniesContext()', () => {
    test('Default state', () => {
      let actual

      const SomeComponent = () => {
        const { state } = useMyCompaniesContext()
        actual = state
        return null
      }

      mount(
        <useMyCompaniesContext.Provider>
          <SomeComponent />
        </useMyCompaniesContext.Provider>,
      )

      expect(actual).toEqual({
        lists: [],
        filter: '',
        sortBy: 'recent',
        selectedIdx: 0,
      })
    })

    test('Provided state', () => {
      let actual

      const SomeComponent = () => {
        const { state } = useMyCompaniesContext()
        actual = state
        return null
      }

      mount(
        <useMyCompaniesContext.Provider {...initialState}>
          <SomeComponent />
        </useMyCompaniesContext.Provider>,
      )

      expect(actual).toEqual({
        ...initialState,
        lists: _.orderBy(initialState.lists, 'name'),
      })
    })
  })
})
