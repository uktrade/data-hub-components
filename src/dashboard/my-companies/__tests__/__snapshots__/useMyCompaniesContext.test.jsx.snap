// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store useMyCompaniesContext() Provider matches snapshot 1`] = `<Provider />`;
